src/main/java/com/rapid7/client/dcerpc/io/PacketOutput.java

```diff
@@ -100,29 +110,48 @@ public void writeString(final String string, final boolean nullTerminate)
 
         if (string != null) {
             writeReferentID();
-            writeInt(maximumChars);
-            writeInt(0);
-            writeInt(currentChars);
-            writeChars(string);
-
-            if (nullTerminate) {
-                writeShort((short) 0);
-            }
-
-            align();
+            writeString(string, maximumChars, currentChars, nullTerminate);
```

Comment:
This should be calling writeStringRef, and the null check and referent ID are no longer required here.

Reply:
:+1: will do